internal/state/store.go

```diff
@@ -131,12 +139,21 @@ func IsEmpty(store dbStore) (bool, error) {
 
 // NewStore creates the dbStore of the state pkg.
 func NewStore(db dbm.DB, options StoreOptions) Store {
-	return dbStore{db, options}
+	m := NopMetrics()
+	if options.Metrics != nil {
```

Comment:
why do we need the 2nd pointer if we can have a private method `func (store dbStore) metrics { return store.StoreOptions.Metrics }` ?

Reply:
Removed!